### PR TITLE
Added Package.swift

### PR DESCRIPTION
**Pull request by simone-gasparini**
Hi @TimOliver !
I've found a working solution for the SPM,
but I thing that we need to discuss about some modifications in your code.

This is your Swift example App moved to SPM: [TOCropViewControllerSPMExample](https://github.com/DualBit/TOCropViewControllerSPMExample)

### CODE REVIEW

**Comment by TimOliver** [comment]
Hi @simone-gasparini!

Um. Wow! Um, thanks for that!

That's a lot of moving files around and renaming header paths. I was okay doing that in `TORoundedButton` but this library is a lot bigger and a lot more complex. Having to completely change the structure of the library to accommodate a dependency manager isn't really great.

I'd rather wait until SPM is at a point where we can add support without having to do so much moving around. What do you think?

**Comment by simone-gasparini** [comment]
Hi @TimOliver 

in reality no files have been moved, but yes, the header files need to be modified.

I've found that:
- SPM need relative path for .h files, not only the name (sincerily i don't know if this can cause problem with the original project, other package managers and CI)
- `publicHeadersPath` has to be `Include`, even if you specify other path, it search for include 🤷‍♂️
- In the include folder we can specify a [module.modulemap](https://github.com/TimOliver/TOCropViewController/blob/d39c9c659a1c2d17f335dfe224e16de39c759b19/Objective-C/TOCropViewController/Include/module.modulemap) that define the umbrella headers path, so no need to move .h files around (this can do even in `TORoundedButton`)

In conclusion the big change is to move imports from `filename` to `relativefilepath/filename`.

I refer to you because I don't know what problems this can bring.

**Comment by GuillermoCiqueHolvi** [comment]
This would be great!
Relative to if this is worth it or not, SPM is not just "a dependency manager" hopefully this will become "**the** dependency manager" sooner than later. Also, it's impossible to know if and when support for it will be possible without these kind of changes.

**Comment by TimOliver** [comment]
Hey @simone-gasparini! I had a random idea to go check out [my old company Realm, and how they added SPM support](https://github.com/realm/realm-cocoa/blob/master/Package.swift). The people there are absolute wizards, and as expected, they already seem to have got the whole lot working in SPM. It looks like they discovered it's necessary to set two sets of header search paths because the relative file path changes depending if it's Xcode or the command line building it.

It seems there's definitely a way to support SPM without needing to change the header paths, even if it does require manually providing a module map as well. When I get more time, I'm interested to look into this more.

@GuillermoCiqueHolvi Loving the enthusiasm Guillermo! I'm sure it's definitely Apple's goal of making this the standard dependency manager for all of its platforms, but the barrier-to-entry still needs to be significantly lowered before I can start investing the time needed to add it to all of my projects. :)

**Comment by jaerod95** [comment]
This looks like it hasn't been touched in a while, what's the status of SPM? I'm interested to add a Package.swift so at least it would support being built via Accio.

**Comment by TimOliver** [comment]
@jaerod95 Hi Jason! Nothing new at the moment. I'm focusing on other projects now. As you can see, we got it to the point where it's functioning if all the source files are moved around, but I want to avoid that.

I managed to talk to an Apple engineer from the Swift team recently and they said the best way to solve this would be to explicitly use a custom module map (Which we kind of already have here).

Like I wrote above, Realm managed to get full integration with a C++, Objective-C++ and Swift set of source files, so when I have more time, I'll go through their implementation and see if I can work out how they did it.

Any help would be appreciated. :)

**Comment by TimOliver** [comment]
Hey @simone-gasparini! Sorry for the delay!

I THINK I've worked out how to do SPM support properly now. The puzzle piece I was missing was that apparently you can consolidate all of the header files using symlinks in a common `include` directory, which avoids needing to mess with the locations of the real headers.

I'll merge your changes into a separate branch and see if I can apply that on top of what you've already done. Thanks again for kicking this off! :)
